### ISSUE TITLE
 flip_done timed out

### ISSUE THREAD

**Issue by oheid** [RadeonOpenCompute/ROCK-Kernel-Driver#22]
After about 10 minutes I reliably get X freezes with 4.9.0-kfd, after which I can switch once to VT but not back without hanging the system. I did not observe this with stock kernels, including 4.9.0. TearFree is activated in xorg.conf. dmesg output is
```
[  491.868816] [drm:drm_atomic_helper_commit_cleanup_done [drm_kms_helper]] *ERROR* [CRTC:34:crtc-0] flip_done timed out
[  492.380778] [drm:drm_atomic_helper_wait_for_dependencies [drm_kms_helper]] *ERROR* [CRTC:34:crtc-0] flip_done timed out
[  492.380816] [drm:amdgpu_dm_atomic_commit_tail [amdgpu]] *ERROR* dm_dc_surface_commit: acrtc 0, already busy
[  492.383795] [drm:amdgpu_dm_atomic_commit_tail [amdgpu]] *ERROR* dm_dc_surface_commit: acrtc 0, already busy
[  492.389201] ------------[ cut here ]------------
[  492.389239] WARNING: CPU: 15 PID: 0 at drivers/gpu/drm/amd/amdgpu/../display/amdgpu_dm/amdgpu_dm.c:193 dm_pflip_high_irq+0xb4/0x1b0 [amdgpu]
[  492.389240] Modules linked in: nls_iso8859_1 xt_tcpudp ip6t_rpfilter ip6t_REJECT nf_reject_ipv6 ipt_REJECT nf_reject_ipv4 xt_conntrack ip_set nfnetlink ebtable_nat ebtable_broute ip6table_security ip6table_mangle ip6table_nat nf_conntrack_ipv6 nf_defrag_ipv6 nf_nat_ipv6 ip6table_raw iptable_security iptable_mangle iptable_nat nf_conntrack_ipv4 nf_defrag_ipv4 nf_nat_ipv4 nf_nat nf_conntrack iptable_raw ebtable_filter ebtables ip6table_filter ip6_tables iptable_filter sunrpc kvm_amd kvm amdkfd irqbypass crct10dif_pclmul crc32_pclmul ghash_clmulni_intel amd_iommu_v2 aesni_intel amdgpu aes_x86_64 lrw gf128mul glue_helper ablk_helper cryptd snd_hda_codec_generic snd_hda_codec_hdmi ttm snd_hda_intel snd_hda_codec drm_kms_helper snd_hda_core snd_hwdep drm snd_seq snd_seq_device snd_pcm eeepc_wmi asus_wmi
[  492.389264]  joydev sparse_keymap input_leds snd_timer fb_sys_fops snd syscopyarea mxm_wmi sysfillrect sysimgblt sp5100_tco soundcore i2c_piix4 i2c_designware_platform gpio_amdpt gpio_generic wmi i2c_designware_core sch_fq_codel ip_tables x_tables hid_generic hid_cherry usbkbd usbmouse igb ptp uas usb_storage usbhid pps_core serio_raw hid i2c_algo_bit dca autofs4
[  492.389279] CPU: 15 PID: 0 Comm: swapper/15 Not tainted 4.9.0-kfd #3
[  492.389280] Hardware name: System manufacturer System Product Name/CROSSHAIR VI HERO, BIOS 9945 05/19/2017
[  492.389281]  ffff9cba7e9c3d50 ffffffff92399592 0000000000000000 0000000000000000
[  492.389284]  ffff9cba7e9c3d90 ffffffff9206736b 000000c1706a9fa0 ffff9cba7960d000
[  492.389285]  ffff9cba60990000 0000000000000086 000000000000001a ffff9cba70f91d00
[  492.389287] Call Trace:
[  492.389289]  <IRQ> 
[  492.389292]  [<ffffffff92399592>] dump_stack+0x63/0x81
[  492.389294]  [<ffffffff9206736b>] __warn+0xcb/0xf0
[  492.389296]  [<ffffffff9206745d>] warn_slowpath_null+0x1d/0x20
[  492.389326]  [<ffffffffc07627c4>] dm_pflip_high_irq+0xb4/0x1b0 [amdgpu]
[  492.389355]  [<ffffffffc0763c8b>] amdgpu_dm_irq_handler+0x7b/0x110 [amdgpu]
[  492.389379]  [<ffffffffc06dc93c>] amdgpu_irq_dispatch+0x7c/0x1a0 [amdgpu]
[  492.389403]  [<ffffffffc06dd186>] amdgpu_ih_process+0xd6/0x140 [amdgpu]
[  492.389405]  [<ffffffff923a27e9>] ? timerqueue_add+0x59/0xb0
[  492.389428]  [<ffffffffc06dc596>] amdgpu_irq_handler+0x16/0x30 [amdgpu]
[  492.389430]  [<ffffffff920c2541>] __handle_irq_event_percpu+0x81/0x1a0
[  492.389432]  [<ffffffff920c2683>] handle_irq_event_percpu+0x23/0x60
[  492.389433]  [<ffffffff920c26fe>] handle_irq_event+0x3e/0x60
[  492.389435]  [<ffffffff920c5c6d>] handle_edge_irq+0x7d/0x150
[  492.389437]  [<ffffffff9201f01a>] handle_irq+0x1a/0x30
[  492.389439]  [<ffffffff9284992b>] do_IRQ+0x4b/0xd0
[  492.389442]  [<ffffffff92847bc2>] common_interrupt+0x82/0x82
[  492.389442]  <EOI> 
[  492.389444]  [<ffffffff92846b86>] ? native_safe_halt+0x6/0x10
[  492.389446]  [<ffffffff920d75cd>] ? enqueue_hrtimer+0x3d/0x80
[  492.389449]  [<ffffffff92458428>] arch_safe_halt+0x9/0xd
[  492.389450]  [<ffffffff92846cfe>] acpi_safe_halt+0x1e/0x27
[  492.389452]  [<ffffffff92846d27>] acpi_idle_do_entry+0x20/0x39
[  492.389454]  [<ffffffff92459567>] acpi_idle_enter+0x1c4/0x1ec
[  492.389456]  [<ffffffff92669212>] cpuidle_enter_state+0xf2/0x2c0
[  492.389458]  [<ffffffff92669417>] cpuidle_enter+0x17/0x20
[  492.389460]  [<ffffffff920aa7f3>] call_cpuidle+0x23/0x40
[  492.389462]  [<ffffffff920aaa5c>] cpu_startup_entry+0x14c/0x230
[  492.389464]  [<ffffffff9203bc12>] start_secondary+0x142/0x170
[  492.389465] ---[ end trace 5749764e73ca4208 ]---
```

**Comment by oheid** [comment]
Switching TearFree off  seems to avoid this problem.

**Comment by oheid** [comment]
switching off tearfree helped a lot, but there still seems to be a race condition which causes random hangs every few hours or so.

**Comment by oheid** [comment]
This problem appears to be fixed in ROCm-1.6
